Lib/pathlib.py

```diff
@@ -330,8 +330,7 @@ def _parse_path(cls, path):
             elif len(drv_parts) == 6:
                 # e.g. //?/unc/server/share
                 root = sep
-        unfiltered_parsed = [drv + root] + rel.split(sep)
-        parsed = [sys.intern(x) for x in unfiltered_parsed if x and x != '.']
+        parsed = [sys.intern(str(x)) for x in rel.split(sep) if x and x != '.']
```

Comment:
Whoa, we `sys.intern` everything? That seems like a painful memory leak. Would we be better off with a class level `lru_cache`'d function[^1]?

Don't we already know that `rel` will be `str` here? If it's bytes, `str(x)` is the wrong conversion anyway. Maybe the intern function should be `cls._flavour.str_and_intern`?

[^1]: `@lru_cache(max_size=REASONABLE_VALUE) def intern(x): return x`

Reply:
The interning of the path parts is longstanding pathlib behaviour. Honestly I don't know enough about the benefits/drawbacks of interning to say whether it's reasonable.

> Don't we already know that `rel` will be `str` here?

We know that it's a _subclass_ of `str` (and not `bytes`), but we don't know whether it's a true `str` object (required by `sys.intern()`).

Reply:
> Whoa, we `sys.intern` everything? That seems like a painful memory leak. 

Is it really all that bad? `sys.intern()` calls `PyUnicode_InternInPlace()`, not `PyUnicode_InternImmortal()`. An interned string gets deallocated based on its reference count, like any other object. For example:

```pycon
>>> s = sys.intern('spam and eggs')
>>> t = sys.intern('spam and eggs')
>>> s is t
True
>>> id(s)
139835455199152
>>> del s, t
>>> s = sys.intern('spam and eggs')
>>> id(s)
139835455198912
```